src/marqo/tensor_search/tensor_search.py

```diff
@@ -574,39 +626,57 @@ def _add_documents_structured(config: Config, add_docs_params: AddDocsParams, ma
     unsuccessful_docs: List[Tuple[int, MarqoAddDocumentsItem]] = []
     total_vectorise_time = 0
     batch_size = len(add_docs_params.docs)  # use length before deduplication
-    image_repo = {}
+    content_repo = {}
 
     text_chunk_prefix = marqo_index.model.get_text_chunk_prefix(add_docs_params.text_chunk_prefix)
 
     # Deduplicate docs, keep the latest
     docs, doc_ids = config.document.remove_duplicated_documents(add_docs_params.docs)
 
+    # Check if model is Video/Audio. If so, manually set thread_count to 5
+    media_download_thread_count = add_docs_params.image_download_thread_count
+
     with ExitStack() as exit_stack:
-        image_fields = [field.name for field in marqo_index.field_map_by_type[FieldType.ImagePointer]]
+        media_fields = [
+            field.name for field in 
+            marqo_index.field_map_by_type[FieldType.ImagePointer] +
+            marqo_index.field_map_by_type[FieldType.VideoPointer] +
+            marqo_index.field_map_by_type[FieldType.AudioPointer]
+        ]
+
+        media_field_types_mapping = {field.name: field.type for field in 
+            marqo_index.field_map_by_type[FieldType.ImagePointer] +
+            marqo_index.field_map_by_type[FieldType.VideoPointer] +
+            marqo_index.field_map_by_type[FieldType.AudioPointer]
+        }
+        
 
-        if image_fields:
+        if media_fields:
             with RequestMetricsStore.for_request().time(
-                    "image_download.full_time",
-                    lambda t: logger.debug(
-                        f"add_documents image download: took {t:.3f}ms to concurrently download "
-                        f"images for {batch_size} docs using {add_docs_params.image_download_thread_count} threads"
-                    )
+                "media_download.full_time",
+                lambda t: logger.debug(
+                    f"add_documents media download: took {t:.3f}ms to concurrently download "
+                    f"media for {batch_size} docs using {media_download_thread_count} threads"
+                )
             ):
 
-                if '_id' in image_fields:
+                if '_id' in media_fields:
                     raise api_exceptions.BadRequestError(message="`_id` field cannot be an image pointer field.")
 
-                image_repo = exit_stack.enter_context(
-                    add_docs.download_and_preprocess_images(
-                        docs=docs, thread_count=add_docs_params.image_download_thread_count,
-                        tensor_fields=image_fields,
-                        image_download_headers=add_docs_params.image_download_headers,
+                content_repo = exit_stack.enter_context(
+                    add_docs.download_and_preprocess_content(
+                        docs=docs, 
+                        thread_count=media_download_thread_count,
```

Comment:
Similar issue of limiting image thread count as well. See unstructured comment

Reply:
similar comment as well, copying here


```Yeah, we only use one thread count variable. I don't think we can have different thread counts for different modalities. For example a batch has documents containing 3 modalities at the same time. To make different thread counts work, we'd have to split up the documents into categories, then multithread download each modality group one after the other with the corresponding number of threads the user specified for that modality, or perhaps mini group them and dynamically change the number of threads as different modalities come in. Can you think of a better way to implement this idea?

This would require quite a bit of refactoring of our code in this file. Is it super necessary at this point? Perhaps we can do this in a future release?```